chain/test/Test/Pos/Chain/Block/Slog/LastBlkSlots.hs

```diff
@@ -0,0 +1,164 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Test.Pos.Chain.Block.Slog.LastBlkSlots
+       ( tests
+       ) where
+
+import           Universum
+
+import           Data.ByteString (ByteString)
+import qualified Data.List as List
+import qualified Data.Map.Strict as Map
+
+import           Hedgehog (Gen, Property, discover, (===))
+import qualified Hedgehog as H
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import           Pos.Chain.Block.Slog.LastBlkSlots (LastSlotInfo (..), LastBlkSlots)
+import qualified Pos.Chain.Block.Slog.LastBlkSlots as LastBlkSlots
+import           Pos.Core (AddressHash, unsafeAddressHash)
+import           Pos.Core.Chrono (OldestFirst (..))
+import           Pos.Crypto (PublicKey (..))
+
+
+-- Need a small number of public keys, and for tests, they don't have be proper
+-- public keys.
+genPublicKeyAddressHash :: Gen (AddressHash PublicKey)
+genPublicKeyAddressHash = do
+    let pkeys = [ "one", "two", "three", "four", "five", "six", "seven" :: ByteString ]
+    unsafeAddressHash <$> Gen.element pkeys
+
+getLastSlotInfo :: Int -> Gen [LastSlotInfo]
+getLastSlotInfo len = do
+    start <- Gen.word64 $ Range.linear 0 20000
+    end <- Gen.word64 $ Range.linear (start + fromIntegral len) (start + 1000)
+    lsis <- List.sort . List.take len <$> Gen.shuffle [start .. end]
+    mapM (\lsi -> LastSlotInfo lsi <$> genPublicKeyAddressHash) lsis
+
+getNextLastSlotInfo :: [LastSlotInfo] -> Gen LastSlotInfo
+getNextLastSlotInfo xs = do
+    lsi <- case xs of
+                [] -> Gen.word64 $ Range.linear 0 20000
+                _ -> pure $ 1 + List.last (map lsiFlatSlotId xs)
+    LastSlotInfo (lsi + 1) <$> genPublicKeyAddressHash
+
+genLastBlkSlots :: Gen LastBlkSlots
+genLastBlkSlots = do
+    count <- Gen.int $ Range.linear 1 100
+    let lbs = LastBlkSlots.create count
+    full <- Gen.bool
+    len <- Gen.int $ if full
+                        then Range.linear count (count + 10)
+                        else Range.linear 0 (count - 1)
+    LastBlkSlots.updateMany lbs <$> fmap OldestFirst (getLastSlotInfo len)
```

Comment:
Note to self / other reviewers:

Since `getLastSlotInfo` shuffles the `FlatSlotId`s of the `LastSlotInfo`s, this ensures that `updateMany` doesn't expect monotonically increasing `FlatSlotId`s. In practice, I believe they'll always monotonically increase, though.

Reply:
I added c comment to clear that up.